### DISCUSSION TITLE
Queue assetion failure ESP-NOW video

### DISCUSSION THREAD

**Discussion by orbitcoms** [Mair/esp32-course#101] [Course Qeustions] [answered]
Hi Mair,
I am working along with the ESP-NOW video "Auto Registration" video.

I set up a queue like shown in video but get error assert failed: xQueueGenericSend queue.c:820(pxQueue).

Do you know what might cause this or how I can debug it?

Below is code that does the queue assertion. (I see the "Received Wifi Position" OK then it crashes

void on_receive(const uint8_t *mac_addr, const uint8_t *data, int data_len)
{
if(data_len != sizeof(payload_t))
{
  ESP_LOGI(TAG,"Received incorrect payload");
  return;
}
  printf("Received Wifi Position\n");
  payload_t * payload = (payload_t *) data;
  xQueueSend(message_received_queue, &payload, 0);
}

Thanks

**Comment by Mair** [comment, marked as answer]
payload_t * payload = (payload_t *) data;
xQueueSend(message_received_queue, &payload, 0);

You are sending a pointer to a pointer.
should be 
xQueueSend(message_received_queue, payload, 0);

**Comment by orbitcoms** [comment]
Thanks Mair, I did not realize that

**Comment by orbitcoms** [comment]
Mair,

With regard to queue. If more data comes in before previous data was consumed, does the receiver keep being fired after each time it processes one item until all have been consumed ? ie. RTOS handles continual firing the receiver until all data consumed

**Comment by Mair** [comment]
@orbitcoms 
Think of it as 2 loops running side by side. like this
![image](https://user-images.githubusercontent.com/2477839/168408902-bad1d265-83ee-42d6-b3bd-7a5fdbda36e5.png)

will task 1 keep firing even if task 2 has not had a chance to process and get the next item. Yes it will
What happens if the queue gets full? xQueusSend will return an error.
Hope this makes it cleared
